crates/compiler/derive/src/decoding/record/decodeWith.rs

```diff
@@ -0,0 +1,118 @@
+use roc_can::expr::Expr;
+
+use roc_module::called_via::CalledVia;
+
+use roc_module::symbol::Symbol;
+use roc_region::all::Loc;
+use roc_types::subs::{Content, FlatType, RecordFields, SubsSlice, TagExt, UnionTags, Variable};
+use roc_types::types::RecordField;
+
+use crate::synth_var;
+use crate::util::Env;
+
+/// Makes the vars for decoding this particular field and decode format
+fn make_decode_with_vars(
```

Comment:
It would be helpful to list in the doc comment what this is taking in and exactly what it is returning, as the other functions do.